src/errors.rs

```diff
@@ -162,24 +162,65 @@ impl Display for UnusedAuditAsErrors {
     }
 }
 
-#[derive(Debug, Error)]
-#[error("{package}:{version}")]
-pub struct NeedsAuditAsError {
+#[derive(Debug, Error, PartialEq, Eq)]
+#[error("{package}{}", .version.as_ref().map(|v| format!(":{v}")).unwrap_or_default())]
+pub struct PackageError {
     pub package: PackageName,
-    pub version: semver::Version,
+    pub version: Option<VetVersion>,
 }
 
-#[derive(Debug, Error)]
-#[error("{package}:{version}")]
-pub struct ShouldntBeAuditAsError {
-    pub package: PackageName,
-    pub version: semver::Version,
+///////////////////////////////////////////////////////////
+// CratePolicyErrors
+///////////////////////////////////////////////////////////
+#[derive(Debug, Error, Diagnostic, PartialEq, Eq)]
+#[error("There are some issues with your third-party policy entries")]
+#[diagnostic()]
```

Comment:
You don't need this empty `#[diagnostic()]` directive.

Reply:
There's one on `AuditAsErrors` too. I haven't used `miette` before and neglected to look up the macro.

Reply:
Actually, looks like there are a few scattered around.